shell.nix

```diff
@@ -0,0 +1,31 @@
+let
+  moz_overlay = import (builtins.fetchTarball https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz);
+  nixpkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
+  ruststable = (nixpkgs.latest.rustChannels.stable.rust.override { extensions = [ "rust-src" "rls-preview" "rust-analysis" "rustfmt-preview" ];});
+in
+  with nixpkgs;
+  stdenv.mkDerivation {
+    name = "rust";
+    nativeBuildInputs = [ pkgconfig ];
+    buildInputs = [
+      openssl
+      nasm
+      rustup
+      ruststable
+      cmake
+      zlib
+      gnumake
+      gcc
+      readline
+      openssl
+      libxml2
+      curl
+      dbus
+      libpulseaudio
+    ];
+
+    shellHook = ''
+        export OPENSSL_DIR="${openssl.dev}"
+        export OPENSSL_LIB_DIR="${openssl.out}/lib"
+    '';
+  }
```

Comment:
If possible I would like this to be added,
makes getting up and running with development so much easier (happy to also add a README.md section about how to use this if you like)

Reply:
There might be a case for adding it -- although I suspect very few contributors are running NixOs -- but it is absolutely out of scope for this PR.

Reply:
you don't need NixOS at all to use this :) 
works on any linux and OSX with a user-space install

but noted. will remove